File: server/java/src/main/java/com/server/reveal/UserContextProvider.java
package com.server.reveal;

import io.revealbi.core.IRVUserContext;
import io.revealbi.core.RVUserContext;
import io.revealbi.servlet.IRVServletUserContextProvider;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.util.HashMap;

@Component
public class UserContextProvider implements IRVServletUserContextProvider {

    @Value("${SQL_SERVER_HOST:localhost}")
    private String sqlServerHost;

    @Value("${SQL_SERVER_DATABASE:Northwind}")
    private String sqlServerDatabase;

    @Value("${SQL_SERVER_USERNAME:sa}")
    private String sqlServerUsername;

    @Value("${SQL_SERVER_PASSWORD:password}")
    private String sqlServerPassword;

    @Value("${SQL_SERVER_SCHEMA:dbo}")
    private String sqlServerSchema;

    @Override
    public IRVUserContext getUserContext(HttpServletRequest request) {
        String headerValue = request.getHeader("x-header-one");
        String userId = null;
        String orderId = null;

        if (headerValue != null && !headerValue.isEmpty()) {
            String[] pairs = headerValue.split(",");
            for (String pair : pairs) {
                String[] kv = pair.split(":", 2);
                if (kv.length == 2) {
                    String key = kv[0].trim();
                    String value = kv[1].trim();
                    if (key.equalsIgnoreCase("userId")) {
                        userId = value;
                    } else if (key.equalsIgnoreCase("orderId")) {
                        orderId = value;
                    }
                }
            }
        }

        // default to User role
        String role = "User";

        // null is used here just for demo
        if ("BLONP".equals(userId) || userId == null) {
            role = "Admin";
        }

        String[] filterTables = role.equals("Admin")
            ? new String[0]
            : new String[]{"Customers", "Orders"};

        var props = new HashMap<String, Object>();
        props.put("OrderId", orderId != null ? orderId : "");
        props.put("Role", role);
        props.put("Host", sqlServerHost);
        props.put("Database", sqlServerDatabase);
        props.put("Username", sqlServerUsername);
        props.put("Password", sqlServerPassword);
        props.put("Schema", sqlServerSchema);
        props.put("FilterTables", filterTables);

        return new RVUserContext(userId, props);
    }
}

File: server/blazor-server/Reveal/UserContextProvider.cs
using Microsoft.Extensions.Options;
using Reveal.Sdk;
using RevealSdk.Server.Configuration;

namespace RevealSdk.Server.Reveal
{
    public class UserContextProvider : IRVUserContextProvider
    {
        private readonly SqlServerOptions _sqlOptions;

        public UserContextProvider(IOptions<SqlServerOptions> sqlOptions)
        {
            _sqlOptions = sqlOptions.Value;
        }

        IRVUserContext IRVUserContextProvider.GetUserContext(HttpContext aspnetContext)
        {

            // Retrieve the values from the request headers
            string userId = aspnetContext.Request.Headers["x-header-customerId"].FirstOrDefault();
            string orderId = aspnetContext.Request.Headers["x-header-orderId"].FirstOrDefault();
            string employeeId = aspnetContext.Request.Headers["x-header-employeeId"].FirstOrDefault();
            //string tenantId = aspnetContext.Request.Headers["x-header-tenantId"].FirstOrDefault();

            // Initialize role to "User"
            string role = "User";

            // Check the userId and set role to "Admin" if conditions are met
            if (userId == "AROUT" || userId == "BLONP" || userId == null)
            {
                role = "Admin";
            }

            var props = new Dictionary<string, object>() {
                    { "OrderId", orderId },
                    { "EmployeeId", employeeId },
                    { "Role", role },
                    { "Host", _sqlOptions.Host },
                    { "Database", _sqlOptions.Database },
                    { "Username", _sqlOptions.Username },
                    { "Password", _sqlOptions.Password },
                    { "Schema", _sqlOptions.Schema } };

            Console.WriteLine("UserContextProvider: " + userId + " " + orderId + " " + employeeId);
            
            return new RVUserContext(userId, props);
        }
    }
}
